### PR TITLE
Reland "install goldctl in docker build"

### PR DESCRIPTION
**Pull request by dnfield**
Reverts flutter/flutter#50088

The flakes this was meant to fix are still occurring after this change.

### CODE REVIEW

**Review by Piinks — APPROVED** [review]
RSLGTM

File: dev/ci/docker_linux/Dockerfile [hunk 1]
```diff
@@ -124,11 +124,18 @@ RUN $GOBINARY get -u github.com/technosophos/dashing
...
 # Gemfile last edited 1/6/2019 (update to trigger a rebuild of Docker image)
-COPY Gemfile /Gemfile
-COPY Gemfile.lock /Gemfile.lock
+COPY ci/docker_linux/Gemfile /Gemfile
+COPY ci/docker_linux/Gemfile.lock /Gemfile.lock
 
 RUN bundle install --system
+
+# Install goldctl, for Golden testing
+# Last updated 1/8/19 (update to rebuild Dockerfile with latest goldctl)
```
Comment (christopherfujino) [inline dev/ci/docker_linux/Dockerfile]: While you're at it, can you update this date, so we can cache bust and get the newest goldctl?

Comment (dnfield) [reply dev/ci/docker_linux/Dockerfile]: Done.

**Review by christopherfujino — APPROVED** [review]
thanks dan, lgtm

**Comment by christopherfujino** [comment]
> This pull request is not suitable for automatic merging in its current state.
> 
> * The status or check suite tool_tests-general-macos has failed. Please fix the issues identified (or deflake) before re-applying this label.

git fetch timed out, restarting

**Comment by christopherfujino** [comment]
Hmm, the docker build is failing with:

```
Step 40/62 : RUN wget --progress=dot:giga "$GRADLE_URL" -O "${GRADLE_ARCHIVE}"
 ---> Running in be900a858c09
--2020-02-04 20:09:32--  http://services.gradle.org/distributions/gradle-4.4-bin.zip
Resolving services.gradle.org (services.gradle.org)... 
104.18.191.9, 104.18.190.9, 2606:4700::6812:be09, ...
Connecting to services.gradle.org (services.gradle.org)|104.18.191.9|:80... 
connected.
HTTP request sent, awaiting response... 
403 Forbidden
2020-02-04 20:09:32 ERROR 403: Forbidden.
```

Not sure why, I can do this locally...

**Comment by christopherfujino** [comment]
@dnfield if you change line 97 to include `https`, the docker image will build.

File: dev/ci/docker_linux/Dockerfile [hunk 2]
```diff
@@ -94,7 +94,7 @@ RUN touch ~/.android/repositories.cfg
 ENV GRADLE_ROOT="/opt/gradle"
 RUN mkdir -p "${GRADLE_ROOT}"
 ENV GRADLE_ARCHIVE="${GRADLE_ROOT}/gradle.zip"
-ENV GRADLE_URL="http://services.gradle.org/distributions/gradle-4.4-bin.zip"
+ENV GRADLE_URL="https://services.gradle.org/distributions/gradle-4.4-bin.zip"
```
Comment (dnfield) [inline dev/ci/docker_linux/Dockerfile]: @christopherfujino  done

**Comment by dnfield** [comment]
This is very consistently failing on the goldctl flake.  I'll wait for @Piinks' other PR to land and merge it into this one.

**Comment by Piinks** [comment]
Patch is in. This can be updated and should no longer experience the flake. 😄

**Comment by dnfield** [comment]
I'm going to land this on red since it is a CI only change that should only help the health of the tree.

**Pull request merged**